### ISSUE TITLE
Antenna automation is buggy

### ISSUE THREAD

**Issue by Gordon-Dry** [Kerbalism/Kerbalism#535]
GameData/Bluedog_DB/Parts/Antennas/bluedog_solarAntenna.cfg

When I set this part to be automated to extend the antenna when reaching space, it already extends the antenna on launch.

Perhaps the reason is that this part is an antenna AND a solar panel.

See the original config:
```
PART
{
	name = bluedog_solarAntenna
	module = Part
	author = CobaltWolf
MODEL
{
	model = Bluedog_DB/Parts/Antennas/bluedog_solarAntenna
}
	scale = 1
	rescaleFactor = 1
	node_attach = 0.0, 0.0, 0.0, 0.0, 0.0, 1.0
	TechRequired = engineering101
	entryCost = 240
	cost = 300
	category = Communication
	subcategory = 0
	title = M17 Solar Antenna //Vanguard I made orbit on March 17th, 1958
	manufacturer = Bluedog Design Bureau
	description = Wehrner grew tired of losing communications with probes every time they run out of battery. This antenna breaks convention by including a small solar cell directly connected to the antenna. Never run out of juice again! 
	attachRules = 1,1,0,0,1
	mass = 0.005
	dragModelType = default
	maximum_drag = 0.2
	minimum_drag = 0.2
	angularDrag = 2
	crashTolerance = 7
	maxTemp = 2000 // = 3000
	PhysicsSignificance = 1
	bulkheadProfiles = srf
	
	tags = aerial antenna radio signal transmit comm communicat dish vanguard
	
	MODULE
	{
		name = ModuleAnimateGeneric
		animationName = extendAntenna
		isOneShot = false
		startEventGUIName = Extend
		endEventGUIName = Retract
		actionGUIName = Toggle Antenna
		allowAnimationWhileShielded = False
	}
	
	MODULE
	{
		name = ModuleDataTransmitter
		antennaType = DIRECT
		packetInterval = 0.6
		packetSize = 2
		packetResourceCost = 12.0
		requiredResource = ElectricCharge
		DeployFxModules = 0
		antennaPower = 500000
		antennaCombinable = True
	}
	
	MODULE
	{
		name = ModuleDeployableSolarPanel
		sunTracking = false
		raycastTransformName = catchThis
		pivotName = catchThis
		isBreakable = false
		resourceName = ElectricCharge
		chargeRate = 0.1
	}
} 
```

and the excerpt from the MM cache:
```
UrlConfig
{
	parentUrl = Bluedog_DB/Parts/Antennas/bluedog_solarAntenna.cfg
	PART
	{
		name = bluedog_solarAntenna
		module = Part
		author = CobaltWolf
		scale = 1
		rescaleFactor = 1
		node_attach = 0.0, 0.0, 0.0, 0.0, 0.0, 1.0
		TechRequired = engineering101
		entryCost = 240
		cost = 300
		category = Communication
		subcategory = 0
		title = M17 Solar Antenna
		manufacturer = Bluedog Design Bureau
		description = <color=orange>Class 2 Antenna DIRECT.</color> Wehrner grew tired of losing communications with probes every time they run out of battery. This antenna breaks convention by including a small solar cell directly connected to the antenna. Never run out of juice again!
		attachRules = 1,1,0,0,1
		mass = 0.005
		dragModelType = default
		maximum_drag = 0.2
		minimum_drag = 0.2
		angularDrag = 2
		crashTolerance = 7
		maxTemp = 2000
		PhysicsSignificance = 1
		bulkheadProfiles = srf
		tags = aerial antenna radio signal transmit comm communicat dish vanguard
		MODEL
		{
			model = Bluedog_DB/Parts/Antennas/bluedog_solarAntenna
		}
		MODULE
		{
			name = ModuleAnimateGeneric
			animationName = extendAntenna
			isOneShot = false
			startEventGUIName = Extend
			endEventGUIName = Retract
			actionGUIName = Toggle Antenna
			allowAnimationWhileShielded = False
		}
		MODULE
		{
			name = ModuleDataTransmitter
			antennaType = DIRECT
			packetSize = 0.00129335234739616
			requiredResource = ElectricCharge
			DeployFxModules = 0
			antennaPower = 500000
			antennaCombinable = True
			packetResourceCost = 2.20732133955612E-05
			packetInterval = 1
		}
		MODULE
		{
			name = ModuleDeployableSolarPanel
			sunTracking = false
			raycastTransformName = catchThis
			pivotName = catchThis
			isBreakable = false
			resourceName = ElectricCharge
			chargeRate = 0.1
		}
		MODULE
		{
			isEnabled = True
			name = BuoyancyControl
			enabled = true
		}
		MODULE
		{
			name = GeometryPartModule
		}
		MODULE
		{
			name = FARAeroPartModule
		}
		MODULE
		{
			name = FARPartModule
		}
		MODULE
		{
			name = SolarPanelFixer
		}
		MODULE
		{
			name = Reliability
			type = ModuleDataTransmitter
			title = Data Transmitter
			redundancy = Communication
			repair = Engineer
			mtbf = 36288000
			extra_cost = 1.0
			extra_mass = 0.1
			rated_radiation = 0.09
			radiation_decay_rate = 10
		}
		MODULE
		{
			name = PlannerController
			title = transmitter
			considered = true
		}
		MODULE
		{
			name = Reliability
			type = SolarPanelFixer
			title = Solar Panel
			redundancy = Power Generation
			repair = true
			mtbf = 36288000
			extra_cost = 2.5
			extra_mass = 1.0
			rated_radiation = 0.09
			radiation_decay_rate = 10
		}
		MODULE
		{
			name = CNConstellationAntennaModule
		}
		MODULE
		{
			name = ModuleAntennaInfo
		}
	}
}
```

**Comment by gotmachine** [comment]
Our code for antenna automation is shitty and needs to be rewritten, there are multiple cases where it is failing / not targeting the right animation module.

**Comment by Gordon-Dry** [comment]
Even built in INTERNAL antennas of pods can be automated (retracted / deployed).
At least the names of the states could be changed to disabled / enabled.
But should INTERNAL antennas even be toggleable?

**Comment by lordcirth** [comment]
Considering that having a fixed antenna anywhere on a craft/station can *decrease* the data rate, it would be very nice to allow turning them off. But perhaps this does not apply to internal ones, as they seem to not transmit science at all.

**Comment by gotmachine** [comment]
Should be fixed in 31f34ee30ceefa1cb9588e109bdd667652686f76
